packages/loader/container-loader/src/deltaManager.ts

```diff
@@ -397,6 +397,8 @@ export class DeltaManager
         } else {
             return {
                 ...common,
+                // last connection mode right before disconnect
+                lastConnectionMode: this.client.mode,
```

Comment:
Interesting. I'd just call it "connectionMode" and move it to common properties.
We record (as part of event) connectionState, so we know that it's disconnect state and I believe humans will have easy time figuring out how to interpret mode when in disconnected state.

Reply:
Isn't `this.client.mode` the requested mode and not the mode in the connection? In other words, it can be different from `this.connection.mode` right? For example, client may ask for "write" mode but the connection established with the "server" is 'read"?
@vladsud

Reply:
There is this really weird line that makes it work, I think:

```typescript
                    this.client.mode = requestedMode;
                    connection = await docService.connectToDeltaStream(this.client);
```

But I agree that this is weird and can easily break.
Also it's requested mode, which can be "read", but storage can return "write".

So yes, I believe we need to add a variable to track it properly, and not rely on this.client.mode.